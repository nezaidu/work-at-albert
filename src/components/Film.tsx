import moment from 'moment';
import React, { FC } from 'react';
import { Text, View } from 'react-native';
import { Film as FilmType, Person } from '../generated/type';


interface FilmProps {
  film: FilmType;
}

const Film: FC<FilmProps> = ({ film }) => {
  return (
    <View style={{ flex: 1, backgroundColor: 'wheat'}}>
      <Text>
        {film.title}
      </Text>
      <Text>
        {film.episodeID}
      </Text>
      <Text>
        {moment(film.releaseDate).local().calendar()}
      </Text>
      {film.characterConnection?.characters?.map((character: Person) => (
        <View>
          <Text>{character.name}</Text>
          <Text>{character.species?.name}</Text>
          <Text>{character.gender === 'n/a' ? 'unknown' : character.gender}</Text>
        </View>
      ))}
    </View>
  )
}


export default Film;
