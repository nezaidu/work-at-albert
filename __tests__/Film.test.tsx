/**
 * @format
 */
import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import {render} from '@testing-library/react-native';
import filmJSON from '../__mocks__/film.json';
import {Film as FilmType} from '../src/generated/type';
import Film from '../src/components/Film';

const film: FilmType = filmJSON as unknown as FilmType;

it('render film', () => {
  const {toJSON} = render(
    <Film
      film={film}
    />,
  );

  const tree = toJSON();

  expect(tree).toMatchSnapshot();
});
