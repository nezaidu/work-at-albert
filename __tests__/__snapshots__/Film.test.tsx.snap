// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render film 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#010103",
              "fontFamily": "Star Jedi Rounded",
              "fontSize": 22,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "lineHeight": 20,
              "paddingBottom": 0,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 3,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#010103",
              "fontFamily": "SerifGothic LT",
              "fontSize": 16,
              "fontStyle": "normal",
              "fontWeight": "normal",
              "lineHeight": 14,
              "paddingLeft": 10,
              "paddingTop": 3,
            },
          ]
        }
      >
        Today at 5:31 PM
      </Text>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#010103",
          "fontFamily": "StarJedi Special Edition",
          "fontSize": 22,
          "fontStyle": "normal",
          "fontWeight": "normal",
          "lineHeight": 18,
          "marginTop": 10,
          "paddingBottom": 0,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
      ]
    }
  >
    Cast
  </Text>
</View>
`;
